From 08bdac8dff053868074871b3bc6318d6d95d476e Mon Sep 17 00:00:00 2001
From: Magnus Damm <damm@opensource.se>
Date: Wed, 28 Mar 2012 19:22:54 +0900
Subject: ARM: mach-shmobile: Use 0x3400 as INTCS vector offset

Update mach-shmobile to use 0x3400 as INTCS_VECT_BASE.

Since the ARM architecture a little while back added support
for 10 bit irqs we can now undo the previously merged commit
9b7c23adb350a108737a993c9c781463c1439dc6 and use 0x3400 as
INTCS vector base.

This change is necessary to avoid overlapping of interrupt
ranges so separate IRQ domains can be used for different INTC
instances. Without this fix the vectors used by various INTC
instances are overlapping on for instance sh7372 which works
at the moment but breaks upcoming IRQ domain support.

Signed-off-by: Magnus Damm <damm@opensource.se>
Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
(cherry picked from commit 1ee8299a9ec1ce5137a044c7768293007b9a3267)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 arch/arm/mach-shmobile/include/mach/irqs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-shmobile/include/mach/irqs.h b/arch/arm/mach-shmobile/include/mach/irqs.h
index 4e686cc2..06a5da3 100644
--- a/arch/arm/mach-shmobile/include/mach/irqs.h
+++ b/arch/arm/mach-shmobile/include/mach/irqs.h
@@ -7,7 +7,7 @@
 #define gic_spi(nr)		((nr) + 32)
 
 /* INTCS */
-#define INTCS_VECT_BASE		0x2200
+#define INTCS_VECT_BASE		0x3400
 #define INTCS_VECT(n, vect)	INTC_VECT((n), INTCS_VECT_BASE + (vect))
 #define intcs_evt2irq(evt)	evt2irq(INTCS_VECT_BASE + (evt))
 
-- 
1.8.0.197.g5a90748

